### ISSUE TITLE
[generator]: do not keep a class under `Azure.Core` namespace even if its accessibility is `internal`

### ISSUE THREAD

**Issue by archerzz** [Azure/autorest.csharp#4597]
## Description

`communication/Communication.JobRouter` has marked some operations `internal`: https://github.com/Azure/azure-rest-api-specs/blob/c4e661cdf92c8f579574008d0cd11874cc303da0/specification/communication/Communication.JobRouter/client.tsp#L25

Thus, the accessibility of some foundation models are marked as `internal`.
![image](https://github.com/Azure/autorest.csharp/assets/4516/1b9d045a-c024-43f3-b9ee-46770633cfe5)

In current implementation, we'll keep a model as long as its accessibility is set. I think we should not keep any foundation model, no matter what the accessibility is. If it's really used, it should be referenced from `Azure.Core`, not from our generated codes.
